### PR TITLE
Issue #3094453 by SV: Prevent error: Call to undefined method ContentBuilder::t()

### PR DESCRIPTION
**Pull request by volodymyr-sydor-lemberg**
## Problem
The website encountered an unexpected error. Please try again later.
Error: Call to undefined method Drupal\social_content_block\ContentBuilder::t() in Drupal\social_content_block\ContentBuilder->getTopics() (line 137 of profiles/contrib/social/modules/social_features/social_content_block/src/ContentBuilder.php).

## Solution
Add & use StringTranslationTrait in the Drupal\social_content_block\ContentBuilder

## Issue tracker

- https://www.drupal.org/project/social/issues/3094453
- https://getopensocial.atlassian.net/browse/YANG-1817

### CODE REVIEW

File: modules/social_features/social_content_block/src/ContentBuilder.php [hunk 1]
```diff
@@ -17,6 +18,8 @@
  */
 class ContentBuilder implements ContentBuilderInterface {
 
+  use StringTranslationTrait;
```
Comment (OH17L) [inline modules/social_features/social_content_block/src/ContentBuilder.php]: Add the **string_translation** service to the service arguments of this class.

File: modules/social_features/social_content_block/social_content_block.services.yml [hunk 2]
```diff
@@ -6,4 +6,4 @@ services:
       - {name: config.factory.override, priority: 5}
   social_content_block.content_builder:
     class: Drupal\social_content_block\ContentBuilder
-    arguments: ['@entity_type.manager', '@database', '@module_handler']
+    arguments: ['@entity_type.manager', '@database', '@module_handler', '@string_translation']
```
Comment (OH17L) [inline modules/social_features/social_content_block/social_content_block.services.yml]:
```suggestion
    arguments:
      - '@entity_type.manager'
      - '@database'
      - '@module_handler'
      - '@string_translation'
```

File: modules/social_features/social_content_block/src/ContentBuilder.php [hunk 3]
```diff
@@ -47,11 +51,14 @@ class ContentBuilder implements ContentBuilderInterface {
...
    *   The module handler to invoke the alter hook with.
+   * @param \Drupal\Core\StringTranslation\TranslationInterface $string_translation
+   *   The string translation.
    */
-  public function __construct(EntityTypeManagerInterface $entity_type_manager, Connection $connection, ModuleHandlerInterface $module_handler) {
+  public function __construct(EntityTypeManagerInterface $entity_type_manager, Connection $connection, ModuleHandlerInterface $module_handler, TranslationInterface $string_translation) {
     $this->entityTypeManager = $entity_type_manager;
     $this->connection = $connection;
     $this->moduleHandler = $module_handler;
+    $this->stringTranslation = $string_translation;
```
Comment (OH17L) [inline modules/social_features/social_content_block/src/ContentBuilder.php]:
```suggestion
    $this->setStringTranslation($string_translation);
```

File: modules/social_features/social_content_block/src/ContentBuilder.php [hunk 4]
```diff
@@ -47,11 +51,14 @@ class ContentBuilder implements ContentBuilderInterface {
    *   Database Service Object.
    * @param \Drupal\Core\Extension\ModuleHandlerInterface $module_handler
    *   The module handler to invoke the alter hook with.
+   * @param \Drupal\Core\StringTranslation\TranslationInterface $string_translation
+   *   The string translation.
    */
-  public function __construct(EntityTypeManagerInterface $entity_type_manager, Connection $connection, ModuleHandlerInterface $module_handler) {
+  public function __construct(EntityTypeManagerInterface $entity_type_manager, Connection $connection, ModuleHandlerInterface $module_handler, TranslationInterface $string_translation) {
```
Comment (OH17L) [inline modules/social_features/social_content_block/src/ContentBuilder.php]:
```suggestion
  public function __construct(
    EntityTypeManagerInterface $entity_type_manager,
    Connection $connection,
    ModuleHandlerInterface $module_handler,
    TranslationInterface $string_translation
  ) {
```

**Comment by ronaldtebrake** [comment]
Also cherrypicked to 6 and 7.x

**Pull request merged**